### PR TITLE
Add PGT.Output.Test.Comments parser

### PR DESCRIPTION
**Pull request by epicallan**
* [x] Depends on #194

### CODE REVIEW

File: pgt/src/PGT/Output/Test/Comments.hs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+      validateRowString rowCount =<< Text.space *> ("rows" <|> "row")
+
+      Text.char ')' *> Text.endOfLine $> RowCountMetaComment rowCount
+      where
+        validateRowString :: RowCount -> Text -> Parser ()
+        validateRowString rowCount@(RowCount count) string
+          | count == 0 && string /= "rows" = Err.error "expected (0 rows) but found (0 row)"
+          | count == 1 && string /= "row"  = Err.error "expected (1 row) but found (1 rows)"
+          | count > 1 && string  /= "rows" = Err.error message
```
Comment (dkubb) [inline pgt/src/PGT/Output/Test/Comments.hs:125]: __question:__ Could this be simplified to only test:

```haskell
        validateRowString rowCount@(RowCount count) string
          | count == 1 && string /= "row"  = "expected (1 row) but found (1 rows)"
          | count /= 1 && string /= "rows" = Err.error message
```

Also, the `expected` in `message` could be defined as:

```haskell
"expected (" <> render rowCount <> " rows) but found (" <> showc count <> " row)"
```

File: pgt/src/PGT/Output/Test/Comments.hs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+parseCommentsMeta = parseErrorComment <|> parseRowCount <|> parseUnexpected
+  where
+    parseErrorComment :: Parser MetaComment
+    parseErrorComment = "-- (ERROR)" *> Text.endOfLine $> ErrorMetaComment
+
+    parseRowCount :: Parser MetaComment
+    parseRowCount = do
+      rowCount <- "-- (" *> (convertImpure <$> Text.scientific)
+
+      validateRowString rowCount =<< Text.space *> ("rows" <|> "row")
```
Comment (dkubb) [inline pgt/src/PGT/Output/Test/Comments.hs:117]: __question:__ What happens to the parser if there was a typo, like `"roe"`?

File: pgt/src/PGT/Output/Test/Comments.hs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+        parseUnexpectedFirstChar :: Parser Text
+        parseUnexpectedFirstChar = do
+          void $ "-- " *> Text.satisfy (not . expectedFirstChar)
+
+          fail . ("expected a comment that starts with an alpha-num char but received:\n\n" <>) . convert
+            =<< parseLineChars
+          where
+            expectedFirstChar :: Char -> Bool
+            expectedFirstChar char = Char.isAlphaNum char && char == '('
```
Comment (epicallan) [inline pgt/src/PGT/Output/Test/Comments.hs:94]: Comment: `Char.isAlphaNum char || char == '('`

File: pgt/src/PGT/Output/Test/Comments.hs (hunk 1 above)
Comment (epicallan) [reply pgt/src/PGT/Output/Test/Comments.hs]: ``render rowCount == "( n rows)" || "(1 row)"``, so current version of the message binding is correct. However, I totally agree with the logic simplification and have made an update.

**Review by dkubb — APPROVED** [review]

**Pull request merged**